File: star_seller.c
/**
 * The Star Seller exercise in algorithmics,
 * written in ANSI C.                       [24 lines of code]
 *
 * I just feel in my element with C. It's like I can talk with the CPU,
 * without the headache of assembly. It's soft and comfortable and fascinating.
 */

#include <stdio.h>
#include <stdbool.h>

int main(int argc, char *argv[])
{
    char name[20];
    int  stars;
    bool valid = false;

    while (!valid) {
        printf("What is your name? ");
        if (scanf("%s", name) > 0)
            valid = true;
    }
    valid = false;
    printf("Hello, %s!\n", name);
    while (!valid) {
        printf("How many stars do you want? ");
        scanf("%d", &stars);
        if (stars >= 0 && stars <= 200)
            valid = true;
    }
    for (; stars; stars--)
        putchar('*');
    printf("\nGoodbye, %s\n", name);
    return 0;
}

// This language is case-sensitive.

File: star_seller.go
package main

/**
 * The Star Seller exercise in algorithmics,
 * written in Go.                           [32 lines of code]
 *
 * I hear Go is a pretty fierce competitor to Rust in the world of modern
 * native apps. Anyways, I'm glad that people finally see the good in native
 * and will (maybe) stop hallucinating over Javascript.
 *
 * I read a bit more about Go, it seems to be very I/O focused, a bit like a
 * native hybrid of Python and Java, since it employs a garbage collector and
 * rather simplistic syntax. You would prefer to use it for things like web
 * application servers.
 */

import (
    "bufio"
    "fmt"
    "os"
)

func main() {
    var name  string
    var stars int
    var valid bool
    reader := bufio.NewReader(os.Stdin)

    for !valid {
        fmt.Print("What is your name? ")
        name, _ = reader.ReadString('\n')
        if len(name) > 0 {
            valid = true
        }
    }
    valid = false

    fmt.Println("Hello, ", name, "!")

    for !valid {
        fmt.Print("How many stars do you want? ")
        fmt.Scan(&stars)
        if stars >= 0 && stars <= 200 {
            valid = true
        }
    }

    for stars > 0 {
        fmt.Print("*")
        stars -= 1
    }
    fmt.Println("\nGoodbye, ", name)
}

// This language is case-sensitive.
